### PR TITLE
[flutter_adaptive_scaffold] Compare breakpoints

### PR DESCRIPTION
**Pull request by martijn00**
*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

Sometimes you want to compare a breakpoint to a baseline or another breakpoint to determine which UI to show or how to handle something. This adds operators so you can check those things.

```dart
Breakpoint.activeBreakpointOf(context) > Breakpoints.large;
```

*List which issues are fixed by this PR. You must list at least one issue.*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/flutter_adaptive_scaffold/README.md [hunk 1]
```diff
@@ -128,6 +126,91 @@ Widget build(BuildContext context) {
 These are the set of widgets that are used on a lower level and offer more
 customizability at a cost of more lines of code.
 
+### Breakpoint
+
+A `Breakpoint` controls the responsive behavior at different screens and configurations.
+
+You can either use a predefined Material3 breakpoint or create your own.
```
Comment (gspencergoog) [inline packages/flutter_adaptive_scaffold/README.md:133]:
```suggestion
You can either use a predefined Material breakpoint or create your own.
```
Regular users won't care which version of Material it belongs to.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Comment by martijn00** [comment]
@Renzo-Olivares can you review?

File: packages/flutter_adaptive_scaffold/lib/src/breakpoints.dart [hunk 2]
```diff
@@ -344,4 +345,84 @@ class Breakpoint {
...
+    return Breakpoint.desktop.contains(Theme.of(context).platform);
+  }
+
+  /// Returns true if the current platform is Mobile.
+  static bool isMobile(BuildContext context) {
+    return Breakpoint.mobile.contains(Theme.of(context).platform);
+  }
+
+  // #docregion Breakpoint operators
+  /// Returns true if the current [Breakpoint] is greater than the given [Breakpoint].
```
Comment (Renzo-Olivares) [inline packages/flutter_adaptive_scaffold/lib/src/breakpoints.dart:360]: super nit: for all of these operators, maybe the language should be `Returns true if this [Breakpoint]....`. What do you think?

Comment (martijn00) [reply packages/flutter_adaptive_scaffold/lib/src/breakpoints.dart]: Done

**Review by Renzo-Olivares — APPROVED** [review]
LGTM

**Pull request merged**